pkg/controller/cloud/node_controller.go

```diff
@@ -284,6 +284,16 @@ func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
 			return err
 		}
 
+		providerID, err := cloudprovider.GetInstanceProviderID(cnc.cloud, types.NodeName(curNode.Name))
+		if err == nil {
+			curNode.Spec.ProviderID = providerID
+		} else {
+			// we should attempt to set providerID on curNode, but
+			// we can continue if we fail since we will attempt to set
+			// node addresses given the node name in getNodeAddressesByProviderIDOrName
+			glog.Errorf("%v", err)
```

Comment:
I agree with your reasoning for not making it required since the node name fallback is supported in every current provider as far as I know. 

My concern is why was ProviderID originally something required prior to making a node schedulable in the kubelet and if it's OK to not follow that pattern here.